tests/test_physics/test_ansatz.py

```diff
@@ -355,3 +356,223 @@ def test_bargmann_Abc_to_phasespace_cov_means(self):
+
+        ansatz = DiffOpPolyExpAnsatz(A, b, c)
+        ansatz2 = DiffOpPolyExpAnsatz(2 * A, 2 * b, 2 * c)
+
+        assert ansatz == ansatz
+        assert ansatz2 == ansatz2
+        assert ansatz != ansatz2
+        assert ansatz2 != ansatz
+
+    def test_simplify(self):
+        A, b, c = Abc_triple(5)
+
+        ansatz = DiffOpPolyExpAnsatz(A, b, c)
+        ansatz = ansatz + ansatz
+
+        assert np.allclose(ansatz.A[0], ansatz.A[1])
+        assert np.allclose(ansatz.A[0], A)
+        assert np.allclose(ansatz.b[0], ansatz.b[1])
+        assert np.allclose(ansatz.b[0], b)
+
+        ansatz.simplify()
+        assert len(ansatz.A) == 1
+        assert len(ansatz.b) == 1
+        assert ansatz.c == 2 * c
+
+    def test_simplify_v2(self):
+        A, b, c = Abc_triple(5)
+
+        ansatz = DiffOpPolyExpAnsatz(A, b, c)
+        ansatz = ansatz + ansatz
+
+        assert np.allclose(ansatz.A[0], ansatz.A[1])
+        assert np.allclose(ansatz.A[0], A)
+        assert np.allclose(ansatz.b[0], ansatz.b[1])
+        assert np.allclose(ansatz.b[0], b)
+
+        ansatz.simplify_v2()
+        assert len(ansatz.A) == 1
+        assert len(ansatz.b) == 1
+        assert np.allclose(ansatz.c, 2 * c)
+
+    def test_order_batch(self):
+        ansatz = DiffOpPolyExpAnsatz(
+            A=[np.array([[0]]), np.array([[1]])],
+            b=[np.array([1]), np.array([0])],
+            c=[1, 2],
+        )
+        ansatz._order_batch()  # pylint: disable=protected-access
+
+        assert np.allclose(ansatz.A[0], np.array([[1]]))
+        assert np.allclose(ansatz.b[0], np.array([0]))
+        assert ansatz.c[0] == 2
+        assert np.allclose(ansatz.A[1], np.array([[0]]))
+        assert np.allclose(ansatz.b[1], np.array([1]))
+        assert ansatz.c[1] == 1
+
+    def test_polynomial_shape(self):
+        A, b, _ = Abc_triple(4)
+        c = np.array([[1, 2, 3]])
+        ansatz = DiffOpPolyExpAnsatz(A, b, c)
+        poly_dim, poly_shape = ansatz.polynomial_shape
+        assert np.allclose(poly_dim, 1)
+        assert np.allclose(poly_shape, (3,))
+
+        A1, b1, _ = Abc_triple(4)
+        c1 = np.array([[1, 2, 3]])
+        ansatz1 = DiffOpPolyExpAnsatz(A1, b1, c1)
+
+        A2, b2, _ = Abc_triple(4)
+        c2 = np.array([[1, 2, 3]])
+        ansatz2 = DiffOpPolyExpAnsatz(A2, b2, c2)
+
+        ansatz3 = ansatz1 * ansatz2
+
+        poly_dim, poly_shape = ansatz3.polynomial_shape
+        assert np.allclose(poly_dim, 2)
+        assert np.allclose(poly_shape, (3, 3))
+
+    def test_decompose_ansatz(self):
+        A, b, _ = Abc_triple(4)
+        c = np.random.uniform(-10, 10, size=(1, 3, 3, 3))
+        ansatz = DiffOpPolyExpAnsatz(A, b, c)
+        decomp_ansatz = ansatz.decompose_ansatz()
+        z = np.random.uniform(-10, 10, size=(1, 1))
+        assert np.allclose(ansatz(z), decomp_ansatz(z))
+        assert np.allclose(decomp_ansatz.A.shape, (1, 2, 2))
+
+    def test_decompose_ansatz_batch(self):
```

Comment:
I think you are testing both the batching of the A,b,c and the batching of z here right? Maybe just add a docstring to the test to explain that.